src/worker-helpers.js

```diff
@@ -184,3 +184,56 @@ export function getCLIEngineOptions(type, config, rules, filePath, fileDir, give
 
   return cliEngineConfig
 }
+
+/**
+ * Gets the list of rules used for a lint job
+ * @param  {Object} cliEngine The CLIEngine instance used for the lint job
+ * @return {Map}              A Map of the rules used, rule names as keys, rule
+ *                            properties as the contents.
+ */
+export function getRules(cliEngine) {
+  // Pull the list of rules used directly from the CLIEngine
+  // Added in https://github.com/eslint/eslint/pull/9782
+  if (Object.prototype.hasOwnProperty.call(cliEngine, 'getRules')) {
+    return cliEngine.getRules()
+  }
+
+  // Attempt to use the internal (undocumented) `linter` instance attached to
+  // the CLIEngine to get the loaded rules (including plugin rules).
+  // Added in ESLint v4
+  if (Object.prototype.hasOwnProperty.call(cliEngine, 'linter')) {
+    return cliEngine.linter.getRules()
+  }
+
+  // Older versions of ESLint don't (easily) support getting a list of rules
+  return new Map()
+}
+
+/**
+ * Given an exiting rule list and a new rule list, determines whether there
+ * have been changes.
+ * @param  {[type]} newRules     A map of the new rules
+ * @param  {[type]} currentRules A Map of the current rules
+ * @return {boolean}             Whether or not there were changes
+ */
+export function didRulesChange(currentRules, newRules) {
+  let rulesChanged = false
+  const currentRuleIds = new Set(currentRules.keys())
+  const newRuleIds = new Set(newRules.keys())
+
+  // Check for new rules added since the last time we sent currentRules
+  const newRulesIds = new Set(newRules.keys())
```

Comment:
Is it intentional to have `newRuleIds` and `newRulesIds`?  If so, that's pretty confusing, IMHO.

I like the approach to comparing Sets laid out in this SO answer: https://stackoverflow.com/a/44827922/1435658

```js
areSetsEqual = (a, b) => a.size === b.size && [...a].every(value => b.has(value));
```

Reply:
It's definitely confusing at best, and that's much cleaner. I'll switch it to something like that 😉.